### ISSUE TITLE
ati-drivers fails to build on latest unstable

### ISSUE THREAD

**Issue by matthiasbeyer** [NixOS/nixpkgs#8856]
```
[...]
 13 9573k   13 1292k    0     0  33735      0  0:04:50  0:00:39  0:04:11 68255make[2]: Leaving directory '/tmp/nix-build-t1lib-5.1.2.drv-0/t1lib-5.1.2/lib/t1lib'
../libtool --mode=link \
        gcc  -o libt1.la type1/arith.lo type1/curves.lo type1/fontfcn.lo type1/hints.lo type1/lines.lo type1/objects.lo type1/paths.lo type1/regions.lo type1/scanfont.lo type1/spaces.lo type1/t1io.lo type1/t1snap.lo type1/t1stub.lo type1/token.lo type1/type1.lo type1/util.lo  t1lib/t1finfo.lo t1lib/t1base.lo t1lib/t1delete.lo t1lib/t1enc.lo t1lib/t1env.lo t1lib/t1load.lo t1lib/t1set.lo t1lib/t1trans.lo t1lib/t1aaset.lo t1lib/t1afmtool.lo t1lib/t1outline.lo t1lib/t1subset.lo t1lib/parseAFM.lo  \
         -version-info 6:2:1 \
         -lm -no-undefined -rpath /nix/store/hihrvri72p2iw1w2awzrfmxh5n3czvw2-t1lib-5.1.2/lib
  CC [M]  /tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/firegl_public.o
  CC [M]  /tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/kcl_acpi.o
  CC [M]  /tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/kcl_agp.o
  CC [M]  /tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/kcl_debug.o
....mkdir .libs
...gcc -shared  type1/.libs/arith.o type1/.libs/curves.o type1/.libs/fontfcn.o type1/.libs/hints.o type1/.libs/lines.o type1/.libs/objects.o type1/.libs/paths.o type1/.libs/regions.o type1/.libs/scanfont.o type1/.libs/spaces.o type1/.libs/t1io.o type1/.libs/t1snap.o type1/.libs/t1stub.o type1/.libs/token.o type1/.libs/type1.o type1/.libs/util.o t1lib/.libs/t1finfo.o t1lib/.libs/t1base.o t1lib/.libs/t1delete.o t1lib/.libs/t1enc.o t1lib/.libs/t1env.o t1lib/.libs/t1load.o t1lib/.libs/t1set.o t1lib/.libs/t1trans.o t1lib/.libs/t1aaset.o t1lib/.libs/t1afmtool.o t1lib/.libs/t1outline.o t1lib/.libs/t1subset.o t1lib/.libs/parseAFM.o  -lm  -Wl,-soname -Wl,libt1.so.5 -o .libs/libt1.so.5.1.2
  CC [M]  /tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/kcl_ioctl.o
(cd .libs && rm -f libt1.so.5 && ln -s libt1.so.5.1.2 libt1.so.5)
(cd .libs && rm -f libt1.so && ln -s libt1.so.5.1.2 libt1.so)
creating libt1.la
(cd .libs && rm -f libt1.la && ln -s ../libt1.la libt1.la)
..cp t1lib/t1lib.h .
../libtool --mode=link \
        gcc  -o libt1x.la t1lib/t1x11.lo \
         -version-info 6:2:1 \
         libt1.la  -lX11  -no-undefined -rpath /nix/store/hihrvri72p2iw1w2awzrfmxh5n3czvw2-t1lib-5.1.2/lib 
/tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/kcl_acpi.c: In function 'KCL_ACPI_Slot_No_Hotplug':
/tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/kcl_acpi.c:836:20: warning: unused variable 'pdev' [-Wunused-variable]
    struct pci_dev *pdev = (struct pci_dev *)data;
                    ^
  CC [M]  /tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/kcl_io.o
gcc -shared  t1lib/.libs/t1x11.o  -Wl,--rpath -Wl,/tmp/nix-build-t1lib-5.1.2.drv-0/t1lib-5.1.2/lib/.libs -Wl,--rpath -Wl,/nix/store/hihrvri72p2iw1w2awzrfmxh5n
/tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/firegl_public.c: In function 'kcl_mem_pat_setup':
/tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/firegl_public.c:4474:9: error: implicit declaration of function 'read_
         cr4 = read_cr4();
         ^
/tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/firegl_public.c:4475:9: error: implicit declaration of function 'write
         write_cr4(cr4 & ~X86_CR4_PGE);
         ^
/tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/firegl_public.c: At top level:
/tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/firegl_public.c:6399:12: warning: 'KCL_fpu_save_init' defined but not 
 static int KCL_fpu_save_init(struct task_struct *tsk)
            ^
  4  943M    4 40.7M    0     0  2840k      0  0:05:40  0:00:14  0:05:26 4467kcc1: some warnings being treated as errors
/nix/store/75l0iyr5jansnchxdlavqqr9rlqc69fa-linux-3.18.17-dev/lib/modules/3.18.17/source/scripts/Makefile.build:257: recipe for target '/tmp/nix-build-ati-dri
make[4]: *** [/tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x/firegl_public.o] Error 1
make[4]: *** Waiting for unfinished jobs....
......(cd .libs && rm -f libt1x.so.5 && ln -s libt1x.so.5.1.2 libt1x.so.5)
(cd .libs && rm -f libt1x.so && ln -s libt1x.so.5.1.2 libt1x.so)
creating libt1x.la
(cd .libs && rm -f libt1x.la && ln -s ../libt1x.la libt1x.la)
cp t1lib/t1libx.h .
make[1]: Leaving directory '/tmp/nix-build-t1lib-5.1.2.drv-0/t1lib-5.1.2/lib'
make[1]: Entering directory '/tmp/nix-build-t1lib-5.1.2.drv-0/t1lib-5.1.2/type1afm'
../libtool --mode=compile \
        gcc -c  -g -O2 -DT1LIB_IDENT="\"5.1.2\"" -DGLOBAL_CONFIG_DIR="\"/nix/store/hihrvri72p2iw1w2awzrfmxh5n3czvw2-t1lib-5.1.2/share/t1lib\""  type1afm.c
mkdir .libs
 gcc -c -g -O2 -DT1LIB_IDENT=\"5.1.2\" -DGLOBAL_CONFIG_DIR=\"/nix/store/hihrvri72p2iw1w2awzrfmxh5n3czvw2-t1lib-5.1.2/share/t1lib\" type1afm.c  -fPIC -DPIC -o 
./nix/store/75l0iyr5jansnchxdlavqqr9rlqc69fa-linux-3.18.17-dev/lib/modules/3.18.17/source/Makefile:1381: recipe for target '_module_/tmp/nix-build-ati-drivers
make[3]: *** [_module_/tmp/nix-build-ati-drivers-14.12-3.18.17.drv-0/common/lib/modules/fglrx/build_mod/2.6.x] Error 2
Makefile:145: recipe for target 'sub-make' failed
make[2]: *** [sub-make] Error 2
Makefile:24: recipe for target '__sub-make' failed
make[1]: *** [__sub-make] Error 2
make[1]: Leaving directory '/nix/store/75l0iyr5jansnchxdlavqqr9rlqc69fa-linux-3.18.17-dev/lib/modules/3.18.17/build'
Makefile:88: recipe for target 'kmod_build' failed
make: *** [kmod_build] Error 2
+ exitHandler
+ exitCode=2
+ set +e
+ closeNest
+ '[' 0 -gt 0 ']'
+ '[' -n '' ']'
+ '[' 2 '!=' 0 ']'
+ runHook failureHook
+ local hookName=failureHook
+ shift
+ local var=failureHook
+ [[ failureHook =~ Hook$ ]]
+ var+=s
+ eval 'local -a dummy=("${failureHooks[@]}")'
++ dummy=("${failureHooks[@]}")
++ local -a dummy
+ for hook in '"_callImplicitHook 0 $hookName"' '"${dummy[@]}"'
+ _eval '_callImplicitHook 0 failureHook'
+ local 'code=_callImplicitHook 0 failureHook'
+ shift
++ type -t _callImplicitHook 0 failureHook
+ '[' function = function ']'
+ eval '_callImplicitHook 0 failureHook "$@"'
++ _callImplicitHook 0 failureHook
++ local def=0
++ local hookName=failureHook
++ case "$(type -t $hookName)" in
+++ type -t failureHook
++ '[' -z '' ']'
++ return 0
+ return 0
+ '[' -n '' ']'
+ exit 2
...builder for ‘/nix/store/xj5dhb41pk3qwycnhzi58dr2x28ah521-ati-drivers-14.12-3.18.17.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/qwsn88jn31rl42zqk9k4xx5pw32b1n0g-nixos-15.06pre65267.6ad8fab.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/qwsn88jn31rl42zqk9k4xx5pw32b1n0g-nixos-15.06pre65267.6ad8fab.drv’ failed
```

Reported here as @ttuegel suggested on IRC.

**Comment by ttuegel** [comment]
I don't know anything about the ATI/AMD drivers, but that hardware is common enough that the channel should not update if the drivers fail to build. Perhaps we can add a test?

/cc maintainers @MarcWeber @offlinehacker @geerds

**Comment by matthiasbeyer** [comment]
> Perhaps we can add a test

:+1: 

Would be really nice to have a test for this. The hardware is really not specific. I'm having a MSI R9 270X in my workstation, if that matters at all.

**Comment by vcunat** [comment]
Channels shouldn't be blocked by non-free stuff, IMO, in principle. Technically, Hydra.nixos.org won't even build such, so it isn't really possible to run such tests in there.

**Comment by vcunat** [comment]
Could this be just another instance of #2912?

**Comment by matthiasbeyer** [comment]
Closing as dupplication of #2912 

Can someone please port the labels of this over to the referenced issue?
